rasa_sdk/interfaces.py

```diff
@@ -147,6 +149,54 @@ def copy(self) -> "Tracker":
             self.latest_action_name,
         )
 
+    def last_executed_action_has(self, name, skip=0):
+        last = self.get_last_event_for(
+            "action", exclude=[ACTION_LISTEN_NAME], skip=skip
+        )
+        return last is not None and last["name"] == name
+
+    def get_last_event_for(self, event_type, exclude=[], skip=0):
+        def filter_function(e: Dict):
+            has_instance = e["event"] == event_type
+            excluded = e["event"] == "action" and e["name"] in exclude
+
+            return has_instance and not excluded
+
+        filtered = filter(filter_function, reversed(self.applied_events()))
+        for _ in range(skip):
+            next(filtered, None)
+
+        return next(filtered, None)
+
+    def applied_events(self) -> List[Dict]:
+        """Returns all actions that should be applied - w/o reverted events."""
+
+        def undo_till_previous(event_type, done_events):
```

Comment:
types (especially important as the event_type is not the same as in the rasa code)